### DISCUSSION TITLE
Possible issue in PyOpenCL during long GPU run with PySPH

### DISCUSSION THREAD

**Discussion by Qcellaris** [pyopencl/pyopencl#562] [Troubleshooting]
Dear developers,

I have been working with PySPH for a while now and I started using it on a GPU. PySPH (via the Compyle packages) relies on PyOpenCL for it's GPU functionality. After running the code for a couple of hours on a GPU, the process suddenly is killed with an error message relating to PyOpenCL. I have been trying to figure out what goes wrong but I can't find the issue. I posted the original issue, together with the error log, on the GitHub of PySPH:

https://github.com/pypr/pysph/issues/348

Since the error message refers to PyOpenCL, I just wanted to check if it might be a bug in the PyOpenCL code. If this is not the case, I can narrow down the scope of my investigation and try to dig deeper in the Compyle and PySPH packages that are build on the PyOpenCL package.

Kind regards,

Stephan

**Comment by inducer** [comment]
My read on this:
```
TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
    1. pyopencl._cl.Buffer(context: pyopencl._cl.Context, flags: int, size: int = 0, hostbuf: object = None)

Invoked with: <pyopencl.Context at 0x55d6b37af190 on <pyopencl.Device 'NVIDIA TITAN V' on 'NVIDIA CUDA' at 0x55d67dacdfa0>>, 1, 36892362247512260608
```
is that something inside of PySPH gets out of control and tries to allocate implausibly large amounts of memory, here in the tens of thousands of exabytes. This number is so large that it fails to fit into the 64-bit integer that OpenCL accepts as the size of the allocation (which in turn is massively larger than any available GPU memory). An unsigned integer underflow comes to mind as a possible reason. As such, I think the implausible allocation size is the issue, the `TypeError` itself is just a mostly-unrelated symptom.

**Reply by Qcellaris** [reply]
Thanks so much for your quick reply! I will look into this in more detail.
